components/org.wso2.carbon.identity.oauth.dcr/src/test/java/org/wso2/carbon/identity/oauth/dcr/service/DCRManagementServiceTest.java

```diff
@@ -0,0 +1,289 @@
+
+    private String ownerName = "dummyOwner";
+    private String clientName = "dummyClientName";
+    private List<String> dummyGrantTypes = new ArrayList();
+    private List<String> redirectUris = new ArrayList<>();
+    private String tenantDomain = "dummyTenantDomain";
+    private String userName = "dummyUserName";
+    private String dummyOauthConsumerSecret = "dummyOauthConsumerSecret";
+    private String applicationName;
+
+    private RegistrationRequestProfile mockRegistrationRequestProfile;
+    private ApplicationManagementService mockApplicationManagementService;
+    private OAuthAdminService mockOAuthAdminService;
+
+    @BeforeTest
+    public void getInstanceTest() {
+        dummyGrantTypes.add("code");
+        dummyGrantTypes.add("implicit");
+        dcrManagementService = DCRManagementService.getInstance();
+        assertNotNull(dcrManagementService);
+    }
+
+    @DataProvider(name = "debugStatus")
+    public Object[][] getDebugStatus() {
+        return new Boolean[][]{
+                {true},
+                {false}
+        };
+    }
+
+    @Test
+    public void registerOAuthApplicationExceptionTest() throws DCRException {
+        registerOAuthApplication1();
+
+        mockRegistrationRequestProfile = mock(RegistrationRequestProfile.class);
+        when(mockRegistrationRequestProfile.getOwner()).thenReturn(ownerName);
+        when(mockRegistrationRequestProfile.getClientName()).thenReturn(clientName);
+        when(mockRegistrationRequestProfile.getGrantTypes()).thenReturn(dummyGrantTypes);
+        when(mockRegistrationRequestProfile.getTenantDomain()).thenReturn(tenantDomain);
+
+        String carbonHome = Paths.get(System.getProperty("user.dir"), "target").toString();
+        System.setProperty(CarbonBaseConstants.CARBON_HOME, carbonHome);
+        PrivilegedCarbonContext.startTenantFlow();
+        PrivilegedCarbonContext.getThreadLocalCarbonContext().setTenantDomain(tenantDomain);
+        PrivilegedCarbonContext.getThreadLocalCarbonContext().setUsername(userName);
+        try {
+            dcrManagementService.registerOAuthApplication(mockRegistrationRequestProfile);
+        } catch (IllegalStateException ex) {
+            assertEquals(ex.getMessage(), "ApplicationManagementService is not initialized properly");
+        }
+    }
+
+    @Test
+    public void registerOAuthApplicationWithNullExistingSP() throws NoSuchFieldException,
+            IllegalAccessException, DCRException, IdentityApplicationManagementException {
+        registerOAuthApplication1();
+
+        mockApplicationManagementService = mock(ApplicationManagementService.class);
+
+        DCRDataHolder dcrDataHolder = DCRDataHolder.getInstance();
+        dcrDataHolder.setApplicationManagementService(mockApplicationManagementService);
+
+        assertNotNull(dcrDataHolder);
+
+        try {
+            dcrManagementService.registerOAuthApplication(mockRegistrationRequestProfile);
+        } catch (IdentityException ex) {
```

Comment:
add a fail() as mentioned in an earlier comment. do the same for other places

Reply:
added